File: src/main/java/com/github/penguin418/todolist/model/entity/TodoUserEntity.java
package com.github.penguin418.todolist.model.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.BatchSize;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.AuthorityUtils;
import org.springframework.security.core.userdetails.UserDetails;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@EntityListeners(AuditingEntityListener.class)
public class TodoUserEntity implements UserDetails {
    @Id
    @Column(name = "user_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long userId;

    @Column(name = "user_name")
    private String username;

    @Column(name = "user_password")
    private String password;

    @Column(name = "user_nickname")
    private String nickname;

    @Column(name = "created_datetime")
    @CreatedDate
    private LocalDateTime createdDatetime;

    @BatchSize(size = 100)
    @OneToMany(mappedBy = "todoUser", fetch = FetchType.LAZY)
    @JsonBackReference
    private final List<TodoEntity> todos = new ArrayList<>();

    public TodoUserEntity(Long userId, String username, String password, String nickname) {
        this.userId = userId;
        this.username = username;
        this.password = password;
        this.nickname = nickname;
    }

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return AuthorityUtils.createAuthorityList("ROLE_USER");
    }

    @Override
    public String getPassword() {
        return password;
    }

    @Override
    public String getUsername() {
        return username;
    }

    @Override
    public boolean isAccountNonExpired() {
        return true;
    }

    @Override
    public boolean isAccountNonLocked() {
        return true;
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    public boolean isEnabled() {
        return true;
    }
}

File: src/main/java/com/github/penguin418/todolist/model/entity/TodoUserEntity.kt
package com.github.penguin418.todolist.model.entity

import com.fasterxml.jackson.annotation.JsonBackReference
import jakarta.persistence.*
import lombok.Getter
import org.hibernate.annotations.BatchSize
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.authority.AuthorityUtils
import java.time.LocalDateTime

@Entity
@Getter
@EntityListeners(AuditingEntityListener::class)
class TodoUserEntity(
        @field:GeneratedValue(strategy = GenerationType.IDENTITY)
        @field:Column(name = "user_id")
        @field:Id
        var userId: Long?,
        @field:Column(name = "user_name")
        var username: String,
        @field:Column(name = "user_password")
        var password: String,
        @field:Column(name = "user_nickname")
        var nickname: String?
) {
    constructor(): this(null, "", "", "")

    @Column(name = "created_datetime")
    @CreatedDate
    var createdDatetime: LocalDateTime? = null

    @BatchSize(size = 100)
    @OneToMany(mappedBy = "todoUser", fetch = FetchType.LAZY)
    @JsonBackReference
    var todos: List<TodoEntity> = ArrayList()
    val authorities: Collection<GrantedAuthority?>
        get() = AuthorityUtils.createAuthorityList("ROLE_USER")
    val isAccountNonExpired: Boolean
        get() = true
    val isAccountNonLocked: Boolean
        get() = true
    val isCredentialsNonExpired: Boolean
        get() = true
    val isEnabled: Boolean
        get() = true
}
